incubator/hnc/internal/validators/object_test.go

```diff
@@ -332,3 +333,63 @@ func TestUserChanges(t *testing.T) {
 		})
 	}
 }
+
+func TestCreatingConflictSource(t *testing.T) {
+	// Construct a forest ('a' is the root).
+	// a <- b <- c
+	//   |
+	//   <- d
+	f := foresttest.Create("-aba")
+
+	// Create secret objects in namespace 'c' and 'd'.
+	createSecret("conflict-c", "c", f)
+	createSecret("conflict-d", "d", f)
+
+	o := &Object{Forest: f}
```

Comment:
Better to create this inside `t.Run()` so you get a fresh forest for each test. Otherwise each test run can impact the next test run.

Also - _consider_ having a minimal forest for each test case. E.g. for the first one, the forest could be `-a`, for the second it can be `-ab`, with a way to communicate which namespace should have the secrets. This might be overkill, but the reason I wrote `foresttest` in the first place was so that we didn't have to create an "omni-forest" that could handle every possible test case.

Though in this case, it's still fairly simple so my proposed change might be overkill - your call.

Reply:
sgtm will update it